FBReader/src/main/java/org/geometerplus/android/fbreader/benetech/AddToReadingListDialogActivity.java

```diff
@@ -0,0 +1,269 @@
+        MyBooksPagerAdapter pagerAdapter = new MyBooksPagerAdapter(getSupportFragmentManager());
+        viewPager.setAdapter(pagerAdapter);
+
+        tabLayout = (TabLayout) findViewById(R.id.tab_layout);
+        tabLayout.setTabsFromPagerAdapter(pagerAdapter);
+        tabLayout.setupWithViewPager(viewPager);
+
+        for (int index = 0; index < tabLayout.getTabCount(); ++index) {
+            TabLayout.Tab tab = tabLayout.getTabAt(index);
+            tab.setTag(pagerAdapter.getTabId(index));
+            tab.setCustomView(pagerAdapter.getTabView(index));
+        }
+
+        setCurrentPageToLastSelectedPage(viewPager);
+    }
+
+    private void setCurrentPageToLastSelectedPage(ViewPager viewPager) {
+        int previousSelectedPageIndex = sharedPreferences.getInt(SHARE_PREFERENCE_CURRENT_PAGE_INDEX_TAG, 0);
+        viewPager.setCurrentItem(previousSelectedPageIndex, true);
+    }
+
+    @Override
+    public void onBackPressed() {
+        ViewPager viewPager = (ViewPager) findViewById(R.id.view_pager);
+        int currentTabContainerIndex = viewPager.getCurrentItem();
+        MyBooksPagerAdapter viewPagerAdapter = (MyBooksPagerAdapter) viewPager.getAdapter();
+        AbstractBaseTabContainer currentContainerFragment = (AbstractBaseTabContainer) viewPagerAdapter.getItem(currentTabContainerIndex);
+        boolean isPopFragment = currentContainerFragment.popFragment();
+        if (!isPopFragment) {
+            finish();
+        }
+    }
+
+    /**
+     * Called from child fragment BookshareReadingListsFragment when a reading list is selected
+     */
+    public void onReadingListSelectedWithTitle(String title) {
+        tabLayout.setVisibility(View.GONE);
+        getSupportActionBar().setTitle(title);
+    }
+
+    public void onBookshareReadingListsFragmentAppeared() {
+        tabLayout.setVisibility(View.VISIBLE);
+        getSupportActionBar().setTitle(getTitle());
+    }
+
+    public HashMap<Long, Book> getDownloadedBooksMap() throws Exception{
+        if(bookHashMap == null){
+            bookHashMap = new HashMap<>();
+            String value = Paths.BooksDirectoryOption().getValue();
+            File downloadDir = new File(value);
+            if (!downloadDir.exists())
+                throw new Exception("Download directory does not exist");
+
+            Collection<File> bookFilesFound = FileUtils.listFiles(downloadDir, BOOK_FILE_EXTENSIONS_TO_FILTER_BY, true);
+            SQLiteBooksDatabase database = (SQLiteBooksDatabase) SQLiteBooksDatabase.Instance();
+            for (File bookFile : bookFilesFound) {
+                ZLFile zlFile = ZLFile.createFileByPath(bookFile.getAbsolutePath());
+                final Book book = Book.getByFile(zlFile);
+                if (book != null) {
+                    Date date = database.findLastAccessedDateForBook(book);
+                    book.setLastAccessedDate(date);
+                    if(book.getBookshareId() != 0){
```

Comment:
The if else can be reversed by reversing the if statement logic to if book.getBookshareId() == 0.